### ISSUE TITLE
Use a CA issuer instead of self-signed certificates

### ISSUE THREAD

**Issue by jimmykarily** [epinio/epinio#564]
Creating a CA issuer is very simple: https://cert-manager.io/docs/configuration/selfsigned/#bootstrapping-ca-issuers

and it has the benefit that the user's need only trust the CA once and it is going to work for all the generated certificates. With self-signed certs, one needs to create an exception for each and every cert generated by Epinio (e.g. the different apps).

See also the discussion here: https://github.com/epinio/epinio/issues/476#issuecomment-862468253

Right now we rely on the system-domain in order to decide whether to use let's encrypt certificates or not. This is wrong (see discussion here: https://github.com/epinio/epinio/pull/560#issuecomment-871121956). Ideally the user should be able to change to another issuer even after installing Epinio but this is complicated. For now, it is ok to let the user specify the type of issuer they want to use at installation time. E.g.

```bash
epinio install --tls-private-ca # the default
```

or

```
epinio install --tls-lets-encrypt
```

Being able to use a private CA no matter what the domain is not only more flexible and better UX but it also unblocks us here: https://github.com/epinio/epinio/pull/562#issue-680740923. In other words, a user may want to have a custom domain and still not use let's encrypt for certs.

This story could simply replace self-signed certs with a CA signer or it could also implement the explicit flags above and decouple the code from the `omg.howdoi.website` domain. Let's decide if we want to split in 2 stories or not.

## Acceptance Criteria

The epinio installer CLI accepts a new argument: `--tls-issuer=string`
For now we support `letsencrypt-production` and `selfsigned-issuer`.

The string is a reference to the cert-manager clusterissuer name. A few places in the code will behave differently depending on the type.

* For `letsencrypt-production` the installer sets up Epinio to use letsencrypt. This is the previous 'production' setup.
* For `selfsigned-issuer` the installer sets up Epinio to generate a private CA. This is old the `omg.howdoi.website` setup.

We keep the old defaults, `systemdomain=omg.howdoi.website`
 and `tls=selfsigned-issuer`.

To use the letsencrypt issuer, you must use a public IP and set up DNS for the system domain.
 
There should be a separate flag to control the node port creation for the internal registry (https://github.com/epinio/epinio/blob/main/assets/container-registry/chart/container-registry/templates/container-registry.yaml#L56-L82), something like `--use-node-port-workaround-for-internal-registry`

### Notes

* https://github.com/epinio/epinio/pull/560/files lists all the switches.
* this also needs to add the new args to epinio server deployment.
* we still have the cluster signed certificate from quarks secret for the internal registry.
* with this issue solved, specifying a custom issuer is possible. A follow-up story to make the letsencrypt issuer configurable, so that one can switch to the `dnschallenge`, is no longer needed.
* another follow-up could add support for `external-dns`, to automate the DNS zone management

**Comment by agracey** [comment]
As part of this, I should be able to use the epinio cli (and eventually WebUI?) to export the root CA so I can add it to Chrome's truststore

**Comment by agracey** [comment]
Or it should be documented how to get the root

**Comment by jimmykarily** [comment]
I'd suggest we split this in 2 stories as described at the bottom of the description above.

**Comment by andreas-kupries** [comment]
Places using the magic domain

- https://github.com/epinio/epinio/blob/main/internal/cli/clients/client.go#L1601
- https://github.com/epinio/epinio/blob/main/internal/cli/clients/client.go#L1011
- https://github.com/epinio/epinio/blob/main/internal/api/v1/stage.go#L161
- https://github.com/epinio/epinio/blob/main/internal/auth/certs.go#L54

**Comment by agracey** [comment]
Did I hear you correctly, that this would supersede the need for changing the omg.howdoi.website domain?

**Comment by jimmykarily** [comment]
Not exactly. We will still need something to use as the default domain when the user doesn't specify one. This is mainly for people that try out Epinio (and for demos). But after this is merged: https://github.com/epinio/epinio/pull/590 you will be able to use any domain you want without that affecting what kind of certificates you get. That will allow you to use `xip.io`, `nip.io` or even real domains with both self-signed certificates or lets-encrypts (not sure if lets encrypt works with "magic" domains though).

This issue here is about replacing self signed certificates with private CA ones (a self-signed CA which you only have to trust once).
